common/tree/script.go

```diff
@@ -175,146 +175,152 @@ func (f *MultisigClosure) Script() ([]byte, error) {
 }
 
 func (f *MultisigClosure) Decode(script []byte) (bool, error) {
-	if len(script) == 0 {
-		return false, fmt.Errorf("empty script")
+	tokenizer := txscript.MakeScriptTokenizer(0, script)
+
+	pubkeys := make([]*secp256k1.PublicKey, 0)
```

Comment:
I understand you're using this array to parse the first public key and then use the next opcode to parse either a CHECKSIGADD or CHECKSIG script type.

I like the idea, that's a nice improvement ! But it seems we can simplify this idea by only checking the opcode at `script[1+32]` ?
*  if it's CHECKSIGADD -> use `completeDecodeChecksigAdd`
* if it's CHECKSIGVERIFY or CHECKSIG -> use `completeDecodeChecksig`
* else `return false, nil`

in other words, not sure if we need the tokenizer here. We could make a static check on the 33th byte of the script and move `pubkeys` and `tokenizer` complexity in the private functions.

Maybe I'm missing something, open to suggestions, what do you think ?

Reply:
seams about right :+1: